Support/ceptr/ceptr/species_info.py

```diff
@@ -61,3 +63,254 @@ def __init__(self):
+        symb_list = []
+        lev_list = []
+
+        # Loop over stopping terms first
+        for stp in self.sc_qss_chain_stop:
+            scqssnum = self.dict_qss_species[stp]
+            name_list.append(f"sc_qss[{scqssnum}]")
+            numb_list.append(scqssnum)
+            symb_list.append(stp)
+            lev_list.append(0)
+
+        # Initialize the dataframe
+        scqss_df = pd.DataFrame(
+            {
+                "name": name_list,
+                "number": numb_list,
+                "symbol": symb_list,
+                "level": lev_list,
+            }
+        )
+
+        # Find the list of qss terms that have been filled
+        qss_filled = scqss_df["name"].unique()
+
+        # Loop until all qss terms have been filled
+        level = 1
+        while len(qss_filled) < len(self.qssa_species_list):
+            # Find the sc_qss that only depend upon filled terms
+            for symb in self.qssa_species_list:
+                scqssnum = self.dict_qss_species[symb]
+                if f"sc_qss[{scqssnum}]" in qss_filled:
+                    continue
+                else:
+                    qss_depend = self.dict_qssdepend_scqss[symb]
+
+                    # Determine if the qss species have been previously filled
+                    f_vec = []
+                    for qss_depend in self.dict_qssdepend_scqss[symb]:
+                        f_vec.append(str(qss_depend) in qss_filled)
+
+                    # Only depends upon previously filled entries
+                    if all(f_vec):
+                        name_list.append(f"sc_qss[{scqssnum}]")
+                        numb_list.append(scqssnum)
+                        symb_list.append(symb)
+                        lev_list.append(level)
+
+            scqss_df = pd.DataFrame(
+                {
+                    "name": name_list,
+                    "number": numb_list,
+                    "symbol": symb_list,
+                    "level": lev_list,
+                }
+            )
+            qss_filled = scqss_df["name"].unique()
+
+            level += 1
+
+        scqss_df["sc_dep"] = ""
+        scqss_df["scqss_dep"] = ""
+        # Add in a few more attributes for easy of use
+        for idx, item in scqss_df.iterrows():
+            scqss_df.at[idx, "sc_dep"] = self.dict_qssdepend_sc[item["symbol"]]
+            scqss_df.at[idx, "scqss_dep"] = self.dict_qssdepend_scqss[
+                item["symbol"]
+            ]
+
+        # Return a deepcopy to self
+        self.scqss_df = scqss_df.copy(deep=True)
+
+    def make_sc_dataframe(self):
+        """Make dataframe for sc species."""
+        name_list = []
+        numb_list = []
+        symb_list = []
+
+        for idx in range(self.n_species):
+            name_list.append(f"sc[{idx}]")
+            numb_list.append(idx)
+            symb_list.append(self.nonqssa_species_list[idx])
+
+        # Initialize the dataframe
+        sc_df = pd.DataFrame(
```

Comment:
Can this be a plain `OrderedDict`? That way we don't have to import/install/depend on `pandas`. Just a thought...

Reply:
Chatted with Nick and I am convinced that pandas should stay.